### DISCUSSION TITLE
Error con todos los #include y código base

### DISCUSSION THREAD

**Discussion by Malvin2** [IIC2133-PUC/2024-2#43] [Tarea 0] [answered]
Hola! Tengo un problema, apenas abrí la carpeta de la tarea en vscode me aparece un error en todos los #include que dice:

cannot open source file "X". Please run the 'Select IntelliSense Configuration...' command to locate your system headers.

Y el primer include tambien dice:

#include errors detected. Please update your includePath. Squiggles are disabled for this translation unit (C:\Users\X\OneDrive\Documentos\Universidad\EDD\T0-2024-2-Malvin2\src\puffle\main.c)

Trate de correr el codigo con el primer input de ejemplo del drive para ver si funcionaba al menos y sale esto:

Command: ./puffle input output
==325==
==325== Invalid read of size 4
==325==    at 0x48B9A94: __vfscanf_internal (vfscanf-internal.c:345)
==325==    by 0x48B923C: __isoc99_fscanf (isoc99_fscanf.c:30)
==325==    by 0x10934A: main (main.c:33)
==325==  Address 0xc0 is not stack'd, malloc'd or (recently) free'd

Y tambien mas abajo falta un free pero ni idea de que si es el codigo base.

Alguna idea de como arreglarlo? Estoy super perdido :(

**Comment by tomastrivino** [comment]
Hola
Por el comando que pusiste en valgrind, posiblemente ese error te salió porque el input no existe (no se encuentra el archivo, lo que produce segmentation fault). Puedes probar nuevamente? Respecto a los include asegúrate de tener el archivo .c y .h si vas a hacer módulos.

**Reply by andrius6789** [reply]
Hola yo tengo el mismo problema con los 4 includes :
![image](https://github.com/user-attachments/assets/928eec0c-d705-40f0-980c-6c190b76e0d4)

Por lo que estoy deduciendo vsc esta teniendo problemas en encontrar donde estan las librerias estandars como la de bool en el pc y no se realmente donde estan tampoco. La "solucion" que encontré en internet es utilizar:
```
gcc -I /path/to/system/headers main.c
```
Que realmente no se que tanta ayuda provea si me llegara a setear el path solo para main.c en este caso

**Reply by Malvin2** [reply]
Hola, al final mi solución fue borrar el repo de mi compu, bajarlo de manera diferente y eso lo arreglo.
En un inicio lo que hice fue copiar el repo en mi pc usando git bash en una carpeta y ahí salían todos los errores. Lo borre y en vez de eso lo baje de nuevo con Ubuntu y todo se arreglo. Asumo que como vscode siempre me abre terminal de Ubuntu, en un inicio al haberlo guardado en otra carpeta y con git bash después tenia problemas con los path pero realmente no lo se.

**Comment by tomastrivino** [comment, marked as answer]
Hola,
acabo de compilar tu tarea en mi contenedor sin problemas. **A veces VSCode puede quejarse erróneamente por estas cosas, por lo que la clave está en ver qué es lo que te dice el compilador al ejecutar `make`**. Reitero que el error de memoria que tuviste posiblemente se debe a que no especificaste un archivo de input existente (ya que observo que pusiste `input` como archivo). 

Cualquier cosa cuéntanos por acá
